### PR TITLE
PHOENIX-6989 Expose server side metrics for DDL operations

### PR DESCRIPTION
**Pull request by jinggou**

### CODE REVIEW

File: phoenix-core/src/main/java/org/apache/phoenix/coprocessor/MetaDataEndpointImpl.java [hunk 1]
```diff
@@ -3497,6 +3516,10 @@ public void addColumn(RpcController controller, final AddColumnRequest request,
                     request.getClientVersion(), parentTable, transformingNewTable, addingColumns);
             if (result != null) {
                 done.run(MetaDataMutationResult.toProto(result));
+
+                metricsSource.incrementAlterAddColumnCount();
+                LOGGER.info("Column(s) added successfully, tableName: {}",
```
Comment (jinggou) [inline phoenix-core/src/main/java/org/apache/phoenix/coprocessor/MetaDataEndpointImpl.java:3521]: I'm thinking about logging the names of added columns, but did not found where to get them.

**Comment by virajjasani** [comment]
checkstyle warning [here](https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1638/3/artifact/yetus-general-check/output/results-checkstyle-phoenix-core.txt)

is `RoundRobinResultIteratorIT` successful in local run with this patch?

**Comment by jinggou** [comment]
> 

checkstyle fixed. RoundRobinResultIteratorIT is successful in local run.

**Comment by virajjasani** [comment]
last build was timed out, triggered new one https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1638/6/

**Review by virajjasani — APPROVED** [review]
+1

**Comment by virajjasani** [comment]
@jinggou we don't have PHOENIX-6589 in 5.1 branch, however you could try backporting this to 5.1 branch and as part of the backport, you can also create `MetricsMetadataSource`, `MetricsMetadataSourceFactory`, `MetricsMetadataSourceImpl` etc classes without adding any "schema registry export" related metrics (which were added as part of PHOENIX-6589).

**Comment by virajjasani** [comment]
unless there is any objections, i will merge this PR in 24 hr

**Comment by jinggou** [comment]
UpsertSelectIT is successful in local run

**Pull request merged**